cmd/manager/main.go

```diff
@@ -225,3 +247,90 @@ func checkIfRequiredFilesExist(requiredFiles []string) error {
 	}
 	return nil
 }
+
+// checkConfiguration checks for required configuration in a cluster before starting the controller.
+func (c *clusterConfig) checkConfiguration(config *rest.Config) error {
+	// check OpenShift API config client.
+	if err := c.getOpenShiftConfigClient(config); err != nil {
+		return fmt.Errorf("unable to get OpenShift client: %v", err)
+	}
+	// check OpenShift API operator client
+	if err := c.getOpenShiftOperatorClient(config); err != nil {
+		return fmt.Errorf("unable to get OpenShift operator client: %v", err)
+	}
+	// check for valid K8s version of the cluster
+	if err := c.getK8sVersion(); err != nil {
+		return fmt.Errorf("unable to get K8s version: %v", err)
+	}
+	// check for valid network configuration of the cluster
+	if err := c.getClusterNetwork(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// getOpenShiftConfigClient gets the new OpenShift config v1 client
+func (c *clusterConfig) getOpenShiftConfigClient(config *rest.Config) error {
+	// Get openshift api config client.
+	configClient, err := configclient.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("could not create config clientset: %v", err)
+	}
+	c.oclient = configClient
+	return nil
+}
+
+// getOpenShiftOperatorClient gets a new OpenShift operator v1 client
+func (c *clusterConfig) getOpenShiftOperatorClient(config *rest.Config) error {
+	// Get openshift operator client
+	operatorClient, err := operatorv1.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("could not create operator clientset: %v", err)
+	}
+	c.operatorClient = operatorClient
+	return nil
+}
+
+// getK8sVersion checks for valid k8s version in the cluster. It returns an error for all versions not equal
+// to supported major version. This is being done this way, and not by directly getting the cluster version, as OpenShift CI
+// returns version in the format 0.0.x and not the actual version attached to its clusters.
+func (c *clusterConfig) getK8sVersion() error {
```

Comment:
This needs to only check the K8s version. I don't see a reason to store in the struct.

Reply:
So renaming it to `checkK8sVersion()` and not storing it would be more appropriate